main.js

```diff
@@ -1252,25 +1411,32 @@ function setMeta() {
     }
 
     // delete obsolete states
-    objects.getObjectView('system', 'state', {startkey: 'system.host.' + hostname + '.', endkey: 'system.host.' + hostname + '.\u9999', include_docs: true}, (_err, doc) => {
+    objects.getObjectView('system', 'state', {startkey: hostObjectPrefix + '.', endkey: hostObjectPrefix + '.\u9999', include_docs: true}, (_err, doc) => {
         // identify existing states for deletion, because they are not in the new tasks-list
-        const todelete = doc.rows.filter(out1 => !tasks.some(out2 => out1.id === out2._id));
+        let thishostStates = doc.rows;
+        if (!compactGroupController) {
+            thishostStates = doc.rows.filter(out1 => !out1.id.includes(hostObjectPrefix + '.compactgroup'));
```

Comment:
I think `hostObjectPrefix + '.compactgroup'` should be a constant at the top somewhere.

Reply:
no, only one codeplace is using this. in all normal cases the "init()" method will add it to hostObjecPrefix already

Reply:
I did see that a few times at the bottom. Just search for lowercase compactgroup

Reply:
Yes but should only be in „init“ in : lines directly under each other. I check again.

Reply:
And here:
https://github.com/ioBroker/ioBroker.js-controller/blob/428a72c05a9537aacec5ed9ecbe718513c347195/main.js#L973

IMO, using the same magic string twice already warrants a constant.

Reply:
ahhh so you only mean the "compactgroup-postfix" :-) yes will do